server.js

```diff
@@ -103,14 +103,30 @@ app.get("/api/activities", (req,res)=>{
 				obj.activityName = regexArr[2];
 				obj.ts = t.ts;
 				obj.link = t.links[0].url;
-				data[data.length-1].activites.push(obj)
+				//checks if theres any elements in activities if not pushes current activity
+				if(data[data.length-1].activites.length===0){
+					data[data.length-1].activites.push(obj)
+				//if there IS an activity in the array it checks each element if it has the same name
+				}else{
+					//naming loop so I can reference to break later
+					acitivtyLoop:
+					for (let i = 0; i < data[data.length-1].activites.length; i++) {
```

Comment:
you are repeating 
`data[data.length-1].activities`
a lot, I'd replace this with a single variable for the sake of clarity, use a variable like `mostRecentActivities`

Reply:
You can even update the variable as you need to, since you are pushingin new objects